pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -561,7 +561,11 @@ func encodeNetInfo(port types.NetworkPortStatus) *info.ZInfoNetwork {
 }
 
 func encodeSystemAdapterInfo(ctx *zedagentContext) *info.SystemAdapterInfo {
-	dpcl := ctx.devicePortConfigList
+	dpcl := types.DevicePortConfigList{}
+	item, err := ctx.subDevicePortConfigList.Get("global")
+	if err == nil {
+		dpcl = item.(types.DevicePortConfigList)
+	}
```

Comment:
You shouldn't need this; there are different goroutines but the trigger ensures that an info message is always encoded and sent after the last trigger. 
The likelyhood of a write to memory sitting in some cache store buffer for long enough to make us send stale data is extremely unlikely since we there are thousands of instructions executed as part of saving the DPCL in the main thread, triggering the sending, encoding the info message before we go and access ctx.devicePortConfigList.

Reply:
In that case it makes no sense to have a devicePortConfigList in ctx. The Modify function is only called if there is a change, thus the modify function just need to do the trigger.

Reply:
Right. Will remove devicePortConfigList from ctx.

Reply:
Actually handleDPCLImpl compares the incoming DPCL with the one that we have (stored in devicePortConfigList) and only creates the trigger if they are different. We may have to keep the devicePortConfigList at least for this comparison.

Reply:
No, because pubsub doesn't call the Modify handler if the content is the same.
That comparison logic dates back to an ancient pubsub which didn't use AF_UNIX but fsnotify and it resulted in periodic calls even without changes. But all such comparison code can now be deleted.

Reply:
Ok